### PR TITLE
Added debug display for rays/frame

### PR DESCRIPTION
**Pull request by sebastienlagarde**
### Purpose of this PR
image

This PR doesn't have rays/second support yet- this will be added in a separate branch on top of this changeset. 

---
### Release Notes
Sourced heavily from Ionut's hackweek project. Major point of deviation is using a shader to draw results to the screen, instead of doing an AsyncReadback to write to a text GameObject. 

Rebased https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/2811 on HDRP/ReflectionFilter.

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=hdrtr_reflectionfilter_countrays&unity_branch=graphics%2Fraytracing%2Fdxr&automation-tools_branch=add-platform-filter

**Manual Tests**: What did you do?
Verified overlay looked correct and updated as I panned around the screen in the scene view. 
Confirmed that issue with missing area shadows occurred on HDRP/ReflectionFilter without my changes.

**Automated Tests**: What did you setup?
Nothing

Any test projects to go with this to help reviewers?
None

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?
Low- most of the trickiness was in getting texture formats to play nice. 

**Halo Effect**: None, Low, Medium, High?
Low- the shader code that was added does not change existing shader code, and the invocation of the raycount debug is very self-contained.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassRaytracingReflection.hlsl [hunk 1]
```diff
@@ -44,7 +44,7 @@ void ClosestHitMain(inout RayIntersection rayIntersection : SV_RayPayload, Attri
     // Color display for the moment
     rayIntersection.color = diffuseLighting + specularLighting;
 #else
-    rayIntersection.color = bsdfData.color + builtinData.emissiveColor;
+    rayIntersection.color = bsdfData.color * GetInverseCurrentExposureMultiplier() + builtinData.emissiveColor;
```
Comment (anisunity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassRaytracingReflection.hlsl]: @sebastienlagarde here is the fix we talked about

**Comment by nerites** [comment]
Hey folks, thanks for handling this!

The failure I was seeing on my other branch (https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/2900)  was a failure in 2307_ShadowVeryHigh. It seems like this branch is showing the same failure, unless I'm looking at the wrong builders? 

https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=hdrtr_reflectionfilter_countrays-3&unity_branch=trunk&automation-tools_branch=add-platform-filter#

This branch also seems to be failing 1402_StackLitSG_Testers, though my previous branch was not... do you know of that test is unstable?

**Comment by nerites** [comment]
Checked locally and 2307_ShadowVeryHigh passes, but 1402_StackLitSG_Testers does not.
Actual: 
image
Reference:
image
Diff:
image

But it's shader graph stuff so I'm not sure how my changeset could have impacted this?

**Comment by sebastienlagarde** [comment]
Hey, no worry about this test, it is unrelated. Merging

**Pull request merged**